client/src/controllers/PreviewController.ts

```diff
@@ -758,6 +755,88 @@ export class PreviewController extends Controller<HTMLElement> {
     }
   }
 
+  async restoreScrollPosition(newIframe: HTMLIFrameElement): Promise<void> {
+    const isCrossOrigin = { oldIframe: false, newIframe: false };
+    // Do try/catch for each iframe so we know which of the iframes are cross-origin
+    try {
+      isCrossOrigin.oldIframe =
+        !this.iframeTarget.contentWindow?.location.origin;
+    } catch {
+      isCrossOrigin.oldIframe = true;
+    }
+    try {
+      isCrossOrigin.newIframe = !newIframe.contentWindow?.location.origin;
+    } catch {
+      isCrossOrigin.newIframe = true;
+    }
+
+    // Origins mismatch, something has gone wrong, skip scroll restoration.
+    if (isCrossOrigin.oldIframe !== isCrossOrigin.newIframe) {
+      return Promise.resolve();
+    }
```

Comment:
This can happen if e.g. a Django error occurs during `serve_preview()`, so the new iframe is no longer cross-domain (as it comes directly from Django), but the old iframe is a successful cross-domain iframe. And vice versa, e.g. recovering from a Django error (old iframe is the same-domain error, new iframe is a successful cross-domain iframe). In these cases we just skip scroll restoration altogether as there's no point in doing so (the content is entirely different).